src/app/presence-status-handler.ts

```diff
@@ -79,11 +80,98 @@ class PresenceStatus {
     const count = presenceStatusStore.getNotificationCount();
-
-    presenceStatusStore.setStatus(myPresence.category);
+    if (currentPresenceStatus !== myPresence.category) {
+      presenceStatusStore.setStatus(myPresence.category);
+      this.updateSystemTrayPresence();
+    }
     showBadgeCount(count);
-    showSystemTrayPresence(myPresence.category);
+  };
+
+  /**
+   * Shows the badge count
+   *
+   * @param count {number}
+   */
+  public updateSystemTrayPresence = (): void => {
+    const status = presenceStatusStore.getStatus();
+    let tray = presenceStatusStore.getCurrentTray();
+    const backgroundImage = presenceStatusStore.generateImagePath(
+      status,
+      'tray',
+    );
+    if (!backgroundImage) {
+      return;
+    }
+    if (!tray) {
+      tray = initSysTray();
+      logger.info('presence-status-handler: create and save Symphony tray');
+    } else {
+      tray.setImage(backgroundImage);
+      logger.info('presence-status-handler: new Symphony status updated');
+    }
+    const currentStatus = presenceStatusStore.getStatus();
+    const presenceNamespace = 'PresenceStatus';
+    const isMana = !!windowHandler.isMana;
+    const contextMenu = Menu.buildFromTemplate([
+      {
+        label: i18n.t('My presence')(),
+        visible: isMana,
+        submenu: [
+          {
+            label: i18n.t(EPresenceStatus.AVAILABLE, presenceNamespace)(),
+            type: 'checkbox',
+            checked: currentStatus === EPresenceStatus.AVAILABLE,
+            click: (_item) => {
+              this.handlePresenceChange(EPresenceStatus.AVAILABLE);
+            },
+          },
+          {
+            label: i18n.t(EPresenceStatus.BUSY, presenceNamespace)(),
+            type: 'checkbox',
+            checked: currentStatus === EPresenceStatus.BUSY,
+            click: (_item) => {
+              this.handlePresenceChange(EPresenceStatus.BUSY);
+            },
+          },
+          {
+            label: i18n.t(EPresenceStatus.BE_RIGHT_BACK, presenceNamespace)(),
+            type: 'checkbox',
+            checked: currentStatus === EPresenceStatus.BE_RIGHT_BACK,
+            click: (_item) => {
+              this.handlePresenceChange(EPresenceStatus.BE_RIGHT_BACK);
+            },
+          },
+          {
+            label: i18n.t(EPresenceStatus.OUT_OF_OFFICE, presenceNamespace)(),
+            type: 'checkbox',
+            checked: currentStatus === EPresenceStatus.OUT_OF_OFFICE,
+            click: (_item) => {
+              this.handlePresenceChange(EPresenceStatus.OUT_OF_OFFICE);
+            },
+          },
+        ],
+      },
+      {
+        label: i18n.t('Quit Symphony')(),
+        click: () => app.quit(),
+      },
+    ]);
+    tray?.setContextMenu(contextMenu);
+  };
+
+  private handlePresenceChange = (currentStatus: EPresenceStatus) => {
+    const status = {
+      category: currentStatus,
+      statusGroup: '',
+      timestamp: 0,
```

Comment:
timestamp I know it wont be necessary to change it to current time stamp but it will look nice btw, might need it later, for convenience